### ISSUE TITLE
Allow scan on almalinux packages

### ISSUE THREAD

**Issue by roussidis** [aquasecurity/trivy#2342]
Previously as stated in the https://github.com/aquasecurity/trivy/issues/1021#issuecomment-1011890917

Trivy could not detect vulnerabilities on Almalinux since there was no %{MODULARITYLABEL} set. 

Since almalinux 8.6 as stated in that issue https://bugs.almalinux.org/view.php?id=173 almalinux team fixed the ${MODULARITYLABEL}.

It was tested with Almalinux 8.6
```
# cat /etc/os-release
NAME="AlmaLinux"
VERSION="8.6 (Sky Tiger)"
ID="almalinux"
ID_LIKE="rhel centos fedora"
VERSION_ID="8.6"
PLATFORM_ID="platform:el8"
PRETTY_NAME="AlmaLinux 8.6 (Sky Tiger)"
ANSI_COLOR="0;34"
CPE_NAME="cpe:/o:almalinux:almalinux:8::baseos"
HOME_URL="https://almalinux.org/"
DOCUMENTATION_URL="https://wiki.almalinux.org/"
BUG_REPORT_URL="https://bugs.almalinux.org/"

ALMALINUX_MANTISBT_PROJECT="AlmaLinux-8"
ALMALINUX_MANTISBT_PROJECT_VERSION="8.6"

# rpm -q php --qf "%{NAME} %{VERSION} %{MODULARITYLABEL}\n"
php 7.4.30 php:remi-7.4:8060020220607113500:00000000
```

Latest trivy 0.29 was not able to detect vulnerabilities though
```
2022-06-16T16:49:18.261+0200    INFO    Vulnerability scanning is enabled
2022-06-16T16:49:18.261+0200    INFO    Secret scanning is enabled
2022-06-16T16:49:18.261+0200    INFO    If your scanning is slow, please try '--security-checks vuln' to disable secret scanning
2022-06-16T16:49:18.261+0200    INFO    Please see also https://aquasecurity.github.io/trivy/v0.29.0/docs/secret/scanning/#recommendation for faster secret detection
2022-06-16T16:50:01.829+0200    INFO    Detected OS: alma
2022-06-16T16:50:01.829+0200    INFO    Detecting AlmaLinux vulnerabilities...
2022-06-16T16:50:01.835+0200    INFO    Skipped detection of these packages: ["httpd" "httpd-filesystem" "httpd-tools" "mod_http2" "nginx-filesystem" "perl-IO-Socket-SSL" "perl-Mozilla-CA" "perl-Net-SSLeay"] because modular packages cannot
be detected correctly due to a bug in AlmaLinux. See also: https://bugs.almalinux.org/view.php?id=173
2022-06-16T16:50:01.835+0200    INFO    Number of language-specific files: 1
2022-06-16T16:50:01.835+0200    INFO    Detecting gobinary vulnerabilities...

php 7.4 (alma 8.6)

Total: 0 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 0, CRITICAL: 0)


usr/local/bin/php-security-checker (gobinary)

Total: 0 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 0, CRITICAL: 0)

```

Since the issue was fixed could we enable please the scanning of Almalinux packages once again ?

**Comment by knqyf263** [comment]
Thanks for sharing! We will add support for it shortly.

**Comment by MaineK00n** [comment]
It says to set the appropriate %{MODULARITYLABEL} to the module for 8.6 on 5/24 and rebuild.
https://bugs.almalinux.org/view.php?id=173

Not all modules available in Alma Linux 8.6 will have %{MODULARITYLABEL} set, but the el8.6 package is rebuilt.
Also, %{MODULARITYLABEL} was set for php installed from Remi's Repository.
```console
$ docker run --rm -it almalinux:8 /bin/bash
[root@e186a8ae26e0 /]# cat /etc/os-release 
NAME="AlmaLinux"
VERSION="8.6 (Sky Tiger)"
ID="almalinux"
ID_LIKE="rhel centos fedora"
VERSION_ID="8.6"
PLATFORM_ID="platform:el8"
PRETTY_NAME="AlmaLinux 8.6 (Sky Tiger)"
ANSI_COLOR="0;34"
CPE_NAME="cpe:/o:almalinux:almalinux:8::baseos"
HOME_URL="https://almalinux.org/"
DOCUMENTATION_URL="https://wiki.almalinux.org/"
BUG_REPORT_URL="https://bugs.almalinux.org/"

ALMALINUX_MANTISBT_PROJECT="AlmaLinux-8"
ALMALINUX_MANTISBT_PROJECT_VERSION="8.6"

[root@e186a8ae26e0 /]# dnf install -y php
[root@e186a8ae26e0 /]# rpm -q php --qf "%{NAME} %{VERSION}-%{RELEASE} %{MODULARITYLABEL} %{VENDOR}\n"
php 7.2.24-1.module_el8.3.0+2010+7c76a223 (none) AlmaLinux
[root@e186a8ae26e0 /]# rpm -q httpd --qf "%{NAME} %{VERSION}-%{RELEASE} %{MODULARITYLABEL} %{VENDOR}\n"
httpd 2.4.37-47.module_el8.6.0+2872+fe0ff7aa.1.alma httpd:2.4:8060020220510105858:9edba152 AlmaLinux
[root@e186a8ae26e0 /]# dnf erase -y php && dnf module reset -y php
[root@e186a8ae26e0 /]# dnf module enable -y php:8.0 && dnf install -y php
[root@e186a8ae26e0 /]# rpm -q php --qf "%{NAME} %{VERSION}-%{RELEASE} %{MODULARITYLABEL} %{VENDOR}\n"
php 8.0.13-2.module_el8.6.0+2739+efabdb8f php:8.0:8060020220406185241:a4870ff1 AlmaLinux
[root@e186a8ae26e0 /]# dnf erase -y php && dnf module reset -y php
[root@e186a8ae26e0 /]# dnf install -y https://rpms.remirepo.net/enterprise/remi-release-8.rpm
[root@e186a8ae26e0 /]# dnf module enable -y php:remi-8.1 && dnf install -y php
[root@e186a8ae26e0 /]# rpm -q php --qf "%{NAME} %{VERSION}-%{RELEASE} %{MODULARITYLABEL} %{VENDOR}\n"
php 8.1.7-1.el8.remi php:remi-8.1:8060020220608084706:00000000 Remi's RPM repository <https://rpms.remirepo.net/>
[root@e186a8ae26e0 /]# rpm -q httpd --qf "%{NAME} %{VERSION}-%{RELEASE} %{MODULARITYLABEL} %{VENDOR}\n"
httpd 2.4.37-47.module_el8.6.0+2872+fe0ff7aa.1.alma httpd:2.4:8060020220510105858:9edba152 AlmaLinux
```

Incidentally, as of 2022/06/17, there seem to be no modules in AlmaLinux 9 ......
```console
$ docker run --rm -it almalinux:9 /bin/bash
[root@641d47085491 /]# cat /etc/os-release 
NAME="AlmaLinux"
VERSION="9.0 (Emerald Puma)"
ID="almalinux"
ID_LIKE="rhel centos fedora"
VERSION_ID="9.0"
PLATFORM_ID="platform:el9"
PRETTY_NAME="AlmaLinux 9.0 (Emerald Puma)"
ANSI_COLOR="0;34"
LOGO="fedora-logo-icon"
CPE_NAME="cpe:/o:almalinux:almalinux:9::baseos"
HOME_URL="https://almalinux.org/"
DOCUMENTATION_URL="https://wiki.almalinux.org/"
BUG_REPORT_URL="https://bugs.almalinux.org/"

ALMALINUX_MANTISBT_PROJECT="AlmaLinux-9"
ALMALINUX_MANTISBT_PROJECT_VERSION="9.0"
REDHAT_SUPPORT_PRODUCT="AlmaLinux"
REDHAT_SUPPORT_PRODUCT_VERSION="9.0"
[root@641d47085491 /]# dnf module list
AlmaLinux 9 - AppStream                                            3.3 MB/s | 6.6 MB     00:02    
AlmaLinux 9 - BaseOS                                               1.2 MB/s | 1.9 MB     00:01    
AlmaLinux 9 - Extras                                                10 kB/s |  12 kB     00:01    
[root@641d47085491 /]#
```
